### ISSUE TITLE
Provide a button to easily capture a performance profile

### ISSUE THREAD

**Issue by dpalmeiro** [mozilla-mobile/fenix#14854]
On Firefox Desktop, there is already a very easy way to capture a performance profile by anybody and this really helps investigate problems people have in the wild.

On Fenix, the only way to capture a performance profile is through remote debugging which is quite a heavyweight process and will discourage users to help submit performance problems.  

I would really love to see a profiler button that enables the profiler so that anybody with Fenix can capture and submit these problems.  Ideally, an option under settings that displays/hides a new profiler button on the main UI would be great.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-6569)

**Comment by mcomella** [comment]
I believe the platform work is https://bugzilla.mozilla.org/show_bug.cgi?id=1456606

Triage: we'll leave this in triage until the platform work is done so we don't forget about it but we're hoping for the fenix team to make the UI decisions. CC @liuche

**Comment by mcomella** [comment]
Triage: waiting on platform work then fenix UI.

**Comment by julienw** [comment]
Local symbolication is important for local builds, but less so for official builds.

So an initial MVP could be to capture and directly [upload the result to the storage through our API server](https://github.com/firefox-devtools/profiler-server/blob/master/API.md). The UI should be smart enough to be able to process a raw profile and symbolicate it at load time, taking the symbols from the symbol server.

As a long term solution, these days @mstange is implementing the functionality to run and use a local symbol server to symbolicate from the profiler UI. So I think that with this we may not need the platform work in [bug 1472794](https://bugzilla.mozilla.org/show_bug.cgi?id=1472794) anymore, but of course still [bug 1666226](https://bugzilla.mozilla.org/show_bug.cgi?id=1666226)).

**Comment by mstange** [comment]
> So an initial MVP could be to capture and directly [upload the result to the storage through our API server](https://github.com/firefox-devtools/profiler-server/blob/master/API.md). The UI should be smart enough to be able to process a raw profile and symbolicate it at load time, taking the symbols from the symbol server.

That would be a great start and get us most of the benefits.

> As a long term solution, these days @mstange is implementing the functionality to run and use a local symbol server to symbolicate from the profiler UI.

Yes, this could be used for Fenix builds that use a locally-compiled Gecko. However, in that scenario you usually have a USB connection to your phone already, so the existing method of capturing profiles is usually fine.

> So I think that with this we may not need the platform work in [bug 1472794](https://bugzilla.mozilla.org/show_bug.cgi?id=1472794) anymore

One thing that would be missing with both of these approaches is symbols for Android system libraries. These are (usually) not available on the Mozilla symbol server, and the local symbol server also wouldn't have them, unless it pulled the right file from the device. about:debugging has those symbols because it pulls them through the devtools connection. So, if you need system library symbols, bug 1472794 would still have some benefit - it would substitute system library symbols right on the device.

**Comment by julienw** [comment]
>     So an initial MVP could be to capture and directly upload the result to the storage through our API server. The UI should be smart enough to be able to process a raw profile and symbolicate it at load time, taking the symbols from the symbol server.

Then probably [bug 1715593](https://bugzilla.mozilla.org/show_bug.cgi?id=1715593) (or at least the Gecko side of it) is a dependency, that we could then expose to GeckoView to implement that in Fenix.

**Comment by MarcLeclair** [comment]
For the mobile work week, @mcomella  and I tried to implement a quick MVP. We did manage to get the profile through fenix without adb connection as a JSON string. 

[Here's the .patch for GV / platform.cpp](https://github.com/mozilla-mobile/fenix/files/6889092/get_profile_inapp_profiler.patch.zip).

All that was left was to 1) Gzipped the JSON 2) Send it over to `/compress-store` endpoint of the profiler server (I think that was the correct endpoint) and 3) Fix the UI on the Fenix side to have a button / settings choice.
